TwoNote/app/src/main/java/com/microsoft/device/display/samples/twonote/NoteDetailFragment.kt

```diff
@@ -275,20 +265,46 @@ class NoteDetailFragment : Fragment() {
         }
     }
 
-    private fun startListFragment() {
+    /**
+     * Close NoteDetailFragment after deletion and open either the NoteListFragment (unspanned)
+     * or the GetStartedFragment (spanned)
+     */
+    private fun closeFragment() {
         activity?.let { activity ->
             if (ScreenHelper.isDualMode(activity)) {
+                // Tell NoteListFragment that list data has changed
+                (parentFragmentManager.findFragmentByTag(MainActivity.LIST_FRAGMENT) as NoteListFragment)
+                    .updateArrayAdapter()
+
                 parentFragmentManager.beginTransaction()
                     .replace(
-                        R.id.dual_screen_start_container_id,
-                        NoteListFragment(), null
+                        R.id.dual_screen_end_container_id,
+                        GetStartedFragment(),
+                        null
                     ).commit()
             } else {
                 parentFragmentManager.beginTransaction()
                     .replace(
                         R.id.single_screen_container_id,
-                        NoteListFragment(), null
-                    ).addToBackStack(null).commit()
+                        NoteListFragment(),
+                        null
+                    ).commit()
+            }
+        }
+    }
+
+    // TODO: back gesture does not get overridden by this
+    private fun onBackPressed() {
+        activity?.let {
+            if (ScreenHelper.isDualMode(it)) {
+                parentFragmentManager.beginTransaction()
+                    .replace(R.id.dual_screen_end_container_id, GetStartedFragment(), null)
+                    .commit()
+            } else {
+                parentFragmentManager.beginTransaction()
+                    .replace(R.id.single_screen_container_id, NoteListFragment(), null)
+                    .commit()
+                // it.onBackPressed()
```

Comment:
ooh, looks like some potential infinite recursion. you should uncomment this and see what happens

Reply:
haha it actually refers to the activity's onBackPressed (which is related to the back gesture!)